pyanaconda/ui/gui/spokes/lib/subscription.py

```diff
@@ -0,0 +1,92 @@
+# Helper methods for the Subscription spoke.
+#
+# Copyright (C) 2020  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+from pyanaconda.core.i18n import _
+
+
+def handle_user_provided_value(user_provided_value, valid_values):
+    """Handle user provided value (if any) based on list of valid values.
+
+    There are three possible outcomes:
+    - the value matches one of the valid values, so we preselect the valid value
+    - the value does not match a valid value, so we append a custom value
+      to the list and preselect it
+    - the user provided value is not available (empty string), no matching will be done
+      and no value will be preselected
+
+    :param str user_provided_value: a value provided by user
+    :param list valid_values: a list of valid values
+    :returns: list of values with one value preselected
+    :rtype: list of (str, str, bool) tuples in (<value>, <display string>, <is_preselected>) format
```

Comment:
What about using named tuple. It's really inconvenient to read code with three tuple because you don't know what is what and [2] doesn't really help here.

Or mark this as private method and leave it as it is. I don't mind if it is not an API.

Reply:
Yep, named tuple looks better, let's use it. :)